File: src/main/java/io/awilson/timein/domain/Instructor.java
package io.awilson.timein.domain;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;

import javax.persistence.*;
import java.util.List;

@Data
@Entity
@ToString(exclude = {"students"})
@EqualsAndHashCode(exclude = {"students"})
public class Instructor {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Integer id;
    @Version
    private Integer version;
    private String firstName;
    private String lastName;
    private String email;
    @OneToMany(mappedBy = "instructor")
    private List<Student> students;
    public String getFullName() {
        return firstName + " " + this.lastName;
    }
}

File: src/main/kotlin/io/awilson/timein/domain/Instructor.kt
package io.awilson.timein.domain

import com.fasterxml.jackson.annotation.JsonIgnore
import javax.persistence.*

@Entity
class Instructor {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    var id: Int = 0
    @Version
    var version: Int = 0
    var firstName: String = ""
    var lastName: String = ""
    var email: String = ""
    @OneToMany(mappedBy = "instructor")
    var students: MutableList<Student> = ArrayList()

    val fullName: String
        get() = "$firstName $lastName"
}
